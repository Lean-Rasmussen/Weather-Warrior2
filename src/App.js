import React, { Component } from 'react';
//import styles
import './styles/index.css';
//import components
import NavBar from './components/navigationBar';
import Battle from './components/battle';
import Home from './components/home';
import NewChar from './components/newChar';

class App extends Component {
  constructor(props){
    super(props)
    this.state= {
      activeScreen :''

    }
  }
  setMainScreen(term){
    this.setState({activeScreen: term})
    console.log(this.state.activeScreen)
  };

  render() {
    const setMainScreen=(term) => {this.setMainScreen(term)}

    return (
      <div className="App">
        <h1> Gif warrior 2.0 </h1>
        <h3> time to get giffing</h3>
        <NavBar SetMainScreens ={setMainScreen} />
        <Home />
      </div>
    );
  }
}

export default App;