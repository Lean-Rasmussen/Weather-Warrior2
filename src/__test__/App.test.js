//module import
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
//import components
import NavBar from '../components/navigationBar';
import Home from '../components/home';
import App from '../App';



const app = mount(<App />);
const navBar = mount(<NavBar />);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('App Renders', ()=> {
  expect(app).toMatchSnapshot();
})


it('Navbar renders', ()=> {
  expect(navBar).toMatchSnapshot();
})

it('renders home page correctly',()=>{
  expect(Home).toMatchSnapshot();
});

it("initialises state",()=> {
  expect(app.state().activeScreen).toEqual('')
});


