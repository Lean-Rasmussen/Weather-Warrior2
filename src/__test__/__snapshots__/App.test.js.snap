// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           Gif warrior 2.0 
</h1>,
        <h3>
           time to get giffing
</h3>,
        <NavBar
          SetMainScreens={[Function]}
/>,
        <Home />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Gif warrior 2.0 ",
        },
        "ref": null,
        "rendered": " Gif warrior 2.0 ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " time to get giffing",
        },
        "ref": null,
        "rendered": " time to get giffing",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "SetMainScreens": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             Gif warrior 2.0 
</h1>,
          <h3>
             time to get giffing
</h3>,
          <NavBar
            SetMainScreens={[Function]}
/>,
          <Home />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Gif warrior 2.0 ",
          },
          "ref": null,
          "rendered": " Gif warrior 2.0 ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " time to get giffing",
          },
          "ref": null,
          "rendered": " time to get giffing",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "SetMainScreens": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navbar renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="nav-bar__home"
          id="Home"
          onClick={[Function]}
>
          Home
</button>,
        <div
          className="nav-bar__new-char"
          id="NewChar"
>
          New Char
</div>,
        <div
          className="nav-bar __battle"
          id="Battle"
>
          Battle
</div>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home",
          "className": "nav-bar__home",
          "id": "Home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Home",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Char",
          "className": "nav-bar__new-char",
          "id": "NewChar",
        },
        "ref": null,
        "rendered": "New Char",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Battle",
          "className": "nav-bar __battle",
          "id": "Battle",
        },
        "ref": null,
        "rendered": "Battle",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="nav-bar__home"
            id="Home"
            onClick={[Function]}
>
            Home
</button>,
          <div
            className="nav-bar__new-char"
            id="NewChar"
>
            New Char
</div>,
          <div
            className="nav-bar __battle"
            id="Battle"
>
            Battle
</div>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "nav-bar__home",
            "id": "Home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Home",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Char",
            "className": "nav-bar__new-char",
            "id": "NewChar",
          },
          "ref": null,
          "rendered": "New Char",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Battle",
            "className": "nav-bar __battle",
            "id": "Battle",
          },
          "ref": null,
          "rendered": "Battle",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders home page correctly 1`] = `[Function]`;
